### PR TITLE
Use skylark version of http_file.

### PR DESCRIPTION
**Pull request by xingao267**
Part of #66.

### CODE REVIEW

**Comment by xingao267** [comment]
Fixed.

**Comment by xingao267** [comment]
Saw similar issue here: https://github.com/bazelbuild/bazel/issues/5633. A lot of other rules enforce name extension check, but the downloaded file of skylark http_file will always be `downloaded`. I will have to create a genrule for each of them to rename. Will send a commit for that soon.

File: test/configs/debian8_clang_autoconfig_test.sh [hunk 1]
```diff
@@ -63,7 +63,8 @@ tar -xf ${TEST_TMPDIR}/debian8-clang-${CONFIG_VERSION}-bazel_${BAZEL_VERSION}-au
 rm -rf ${TEST_TMPDIR}/local_config_cc/tools ${TEST_TMPDIR}/local_config_cc/WORKSPACE
 
 # Unpack the tarball containing published toolchain configs for Bazel 0.10.0 from GitHub.
-tar -xf ${TEST_SRCDIR}/bazel_toolchains_test/file/${COMMIT}.tar.gz -C ${TEST_TMPDIR}
+# The downloaded file of `http_file` rule is hardcoded as `file/downloaded`.
+tar -xf ${TEST_SRCDIR}/bazel_toolchains_test/file/downloaded -C ${TEST_TMPDIR}
```
Comment (nlopezgi) [inline test/configs/debian8_clang_autoconfig_test.sh]: I dont think the right solution is to hardcode the downloaded name here, you could instead change the target that this depends on (https://github.com/bazelbuild/bazel-toolchains/blob/master/test/configs/BUILD#L146) from @bazel_toolchains_test//file to third_party/bazel_toolchains_test:rename_file (and create a BUILD file in third_party/bazel_toolchains_test:rename_file that has a genrule to rename).

**Comment by xingao267** [comment]
Ok. So still use native http_file in workspace, except in the repositories.bzl, which is used by bazel_gpg key.

File: WORKSPACE [hunk 2]
```diff
@@ -130,13 +130,13 @@ container_pull(
 http_file(
     name = "debian_docker_gpg",
     sha256 = "1500c1f56fa9e26b9b8f42452a553675796ade0807cdce11975eb98170b3a570",
-    url = "https://download.docker.com/linux/debian/gpg",
+    urls = ["https://download.docker.com/linux/debian/gpg"],
```
Comment (nlopezgi) [inline WORKSPACE]: just to confirm, there is no change here, just swapping using url, to urls, but still the native rule, right?

Comment (xingao267) [reply WORKSPACE]: No change here. native http_file also have urls.

**Pull request merged**